### PR TITLE
fix(elasticloadbalancingv2): can not set sessionTimeout

### LINKED ISSUE #12843: Cannot set sessionTimeout for elbv2.ListenerAction
For `ListenerAction.authenciateOidc()`, `sessionTimeout` is supposed to be a Duration ([docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-elasticloadbalancingv2.AuthenticateOidcOptions.html#sessiontimeout)).

I’m probably missing something really obvious here but 
when I do

```ts
    return elbv2.ListenerAction.authenticateOidc({
       ...,
       sessionTimeout: Duration.seconds(36000)       
    });
```
I get

> `sessionTimeout: “36000” should be a number` 

When I do
```ts
    return elbv2.ListenerAction.authenticateOidc({
       ...,
       sessionTimeout: 36000
    });
```
I get 

> Type 'number' is not assignable to type 'Duration | undefined'

I believe this has to do with a recent change in CloudFormation and a change in CDK ([commit](https://github.com/aws/aws-cdk/pull/11539/commits/f8fffbddf3eeea285a4187f38ade887c1a4478ba)). Two related resource types have a `sessionTimeout` parameter but with different types

1. AWS::ElasticLoadBalancingV2::Listener -> String (changed recently from Integer)
    * => CfnListener construct
1. AWS::ElasticLoadBalancingV2::ListenerRule -> Integer
    * => CfnListenerRule construct

Arguably, it would be better if the type was consistent in CloudFormation but the customer-facing bug is technically in CDK.

### Environment

MonoCDK 1.75

### PR DESCRIPTION
**Pull request by WinterYukky**
## Summary
Application LoadBalancer can not set `sessionTimeout` on `authenticateOidc` except in `defaultActions`. 
This PR fixes this bug.

## Cause of the bug
This is because the CDK uses the same structures for [ListenerRule.AuthenticateOidcConfig](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listenerrule-authenticateoidcconfig.html) and [Listener.AuthenticateOidcConfig](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listener-authenticateoidcconfig.html). These structures have almost the same structure, but validation fails during synthesize because the data type of `sessionTimeout` is different for String and Integer.

```diff
  AuthenticationRequestExtraParams: 
    Key : Value
  AuthorizationEndpoint: String
  ClientId: String
  ClientSecret: String
  Issuer: String
  OnUnauthenticatedRequest: String
  Scope: String
  SessionCookieName: String
- SessionTimeout: String
+ SessionTimeout: Integer
  TokenEndpoint: String
  UseExistingClientSecret: Boolean
  UserInfoEndpoint: String
```

## How to fix?
Add `addRuleAction()` to register an Action for a ListenerRule so that it can hold both config for `Listener` and config for `ListenerRule`. Also, separate `renderActions()` into one for the `Listener` (`defaultActions`) and one for the `ListenerRule` (`actions`) and have them use their own configs.

This allows changes to be made without destroying existing published interfaces.

Closes #12843, #21768.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — DISMISSED** [review]

File: packages/@aws-cdk/aws-elasticloadbalancingv2-actions/.eslintrc.js [hunk 1]
```diff
@@ -1,2 +1,3 @@
 const baseConfig = require('@aws-cdk/cdk-build-tools/config/eslintrc');
+baseConfig.parserOptions.project = __dirname + '/tsconfig.json';
```
[the remark this answers is not in the document]
Comment (WinterYukky) [reply packages/@aws-cdk/aws-elasticloadbalancingv2-actions/.eslintrc.js:2]: The reason behind making this change was that following the error message was displayed during linting.
```
 yarn eslint packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts
yarn run v1.22.19
$ /workspaces/aws-cdk/node_modules/.bin/eslint packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.9.5

Please only submit bug reports when using the officially supported version.

=============

/workspaces/aws-cdk/packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts.
The file must be included in at least one of the projects provided

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Please focus on errors, not warnings. This issue fixed by #7880 . However, `@aws-cdk/aws-elasticloadbalancingv2-actions` is not included then, so I changed it.

[the remark this answers is not in the document]
Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/aws-elasticloadbalancingv2-actions/.eslintrc.js]: Ah, OK. Thanks for letting me know. No problems with this change.

**Comment by TheRealAmazonKendra** [comment]
Will take another look at this tomorrow.

File: packages/@aws-cdk/aws-elasticloadbalancingv2-actions/.eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (WinterYukky) [reply packages/@aws-cdk/aws-elasticloadbalancingv2-actions/.eslintrc.js]: Thank you @TheRealAmazonKendra !
However, all `.eslintrc.js` has removed from individual packages by #24376 🫥. Naturally, I erased it from this PR.

**Review by corymhall — APPROVED** [review]
I can't believe this is necessary (why cloudformation why), but nice solution!

**Review by corymhall — APPROVED** [review]

**Comment by WinterYukky** [comment]
Thank you @corymhall for reviewing this PR 🥰

**Comment by EysaN** [comment]
could someone please clarify the implementation of the fix here because I am still seeing the same behavior in v2.79.0 (build e4a2c01) python 3.11.0
```
action = elbv2.ListenerAction.authenticate_oidc(
     issuer="dummy",
     authorization_endpoint="dummy",
     token_endpoint="dummy",
     user_info_endpoint="dummy",
     client_id="dummy",
     client_secret=SecretValue.secrets_manager("dummy"),
     next=elbv2.ListenerAction.forward([target_group]),
     session_cookie_name="dummy",
     session_timeout=Duration.seconds(14400),
     scope='dummy',
     on_unauthenticated_request=elbv2.UnauthenticatedAction.AUTHENTICATE
 )

**Comment by WinterYukky** [comment]
Thank you @EysaN for your reporting.

This fix makes it possible to apply configuration to different structures by retaining the respective information (numbers and strings) for the `ListenerRule.authenticateOIDCConfig` and `Listener.authenticateOIDCConfig` structures.

I've tried creating a simple Python example with v2.79.0 and it seems like I can Synth without a problem. Can you give me a minimal reproduction to reproduce?

```python
from aws_cdk import (
    Duration,
    SecretValue,
    Stack,
    aws_ec2 as ec2,
    aws_elasticloadbalancingv2 as elbv2,
)
from constructs import Construct

class ExampleStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        vpc = ec2.Vpc(self, "Vpc")
        lb = elbv2.ApplicationLoadBalancer(self, "Lb", 
            vpc=vpc,
        )
        action = elbv2.ListenerAction.authenticate_oidc(
            issuer="dummy",
            authorization_endpoint="dummy",
            token_endpoint="dummy",
            user_info_endpoint="dummy",
            client_id="dummy",
            client_secret=SecretValue.secrets_manager("dummy"),
            next=elbv2.ListenerAction.fixed_response(200,
                content_type="text/plain",
                message_body="Authenticated",
            ),
            session_cookie_name="dummy",
            session_timeout=Duration.seconds(14400),
            scope='dummy',
            on_unauthenticated_request=elbv2.UnauthenticatedAction.AUTHENTICATE
        )
        listener = lb.add_listener("Listener", 
            protocol=elbv2.ApplicationProtocol.HTTP,
            default_action=action,
        )
        listener.add_action("Action", 
            priority=1,
            conditions=[elbv2.ListenerCondition.path_patterns(["/page*"])],
            action=action,
        )
```

**Comment by f1nl0wt3ch** [comment]
Is that something with OIDC, the timeout will be set in String format? I am still facing with this issue with sdk version 2. Anyone solved this problem with typescript?!

`     const authAction = new elbv2_actions.AuthenticateCognitoAction({
        userPool: userpool,
        next: elbv2.ListenerAction.forward([serviceTg]),
        userPoolClient: userPoolClient,
        sessionCookieName: 'dummy',
        sessionTimeout: Duration.hours(1),
        scope: 'openid',
        onUnauthenticatedRequest: elbv2.UnauthenticatedAction.DENY,
        userPoolDomain: userPoolDomain,
      })`

**Comment by f1nl0wt3ch** [comment]
Hi @WinterYukky !

Thank for you answer. I am using v.2.14.0. Let me try again.

**Pull request merged**